pkg/process/util/address.go

```diff
@@ -9,28 +9,48 @@ import (
 	"encoding/binary"
 	"net"
 	"sync"
+
+	"inet.af/netaddr"
 )
 
 // Address is an IP abstraction that is family (v4/v6) agnostic
-type Address interface {
-	Bytes() []byte
-	WriteTo([]byte) int
-	String() string
-	IsLoopback() bool
-	Len() int
+type Address struct {
+	netaddr.IP
 }
 
-// AddressFromNetIP returns an Address from a provided net.IP
-func AddressFromNetIP(ip net.IP) Address {
-	if v4 := ip.To4(); v4 != nil {
-		var a v4Address
-		copy(a[:], v4)
-		return a
+// WriteTo writes the address byte representation into the supplied buffer
+func (a Address) WriteTo(b []byte) int {
+	if a.Is4() {
+		v := a.As4()
+		return copy(b, v[:])
+	}
+
+	v := a.As16()
+	return copy(b, v[:])
+
+}
+
+// Bytes returns a byte slice representing the Address.
+// You may want to consider using `WriteTo` instead to avoid allocations
+func (a Address) Bytes() []byte {
+	if a.Is4() {
```

Comment:
If we want to match the behavior of `net.IP.To4()`, then this conditional should be `a.Is4() || a.Is4in6()`, since `To4()` would implicitly convert IPv4-in-6 to IPv4.

Reply:
Nice catch

Reply:
It is debatable whether we want that implicit conversion, but I wanted to point out the difference.

Reply:
I don't have an opinion, so I updated the code and left a note in the meantime.